nussl/core/audio_signal.py

```diff
@@ -1653,73 +1654,99 @@ def embed_audio(self, ext='.mp3', display=True):
     #          Effect Hooks          #
     ##################################
 
-    def build_effects(self, reset=True, overwrite=False):
+    def apply_effects(self, reset=True, overwrite=False):
         """
-        Builds all effects in the AudioSignal's effects chains using the audio data of the calling 
-        AudioSignal. If reset=True, the current effects chains will be deleted. If overwrite=True, 
-        the calling AudioSignal's audio_data will be overwritten and the stft_data will be deleted. 
+        Applies all effects in the AudioSignal's effects chains using the audio data of the calling 
```

Comment:
I think this needs a little more fleshing out here. There are a few concepts that are not explicitly stated that need to be. Here is the place for us to spell out how the whole effects process works, conceptually. Questions that need to be addressed here: What's an "effects chain"?  How do I put effects in the chain? How are the effects stored/how can I access it to see what's there? Does the order of elements in the chain matter? What steps do I need to take before they're applied? Say it with language first, then talk about the details (what `reset`/`overwrite` do, which are FFmpeg vs SoX, etc), then provide a few bits of example code.

Also, you'll need to add one line of documentation to every effect function saying something to the _effect_ ;-) of "this won't get applied until you call `apply_effects()` and the order matters. for more details see link:`apply_effects()`". It's fine to copy and paste the same text to every effect function both here and in effects.py